demo/cosmic_and_ivm_demo.py

```diff
@@ -1,229 +1,259 @@
 
-    for i, profile in enumerate(cosmic['profiles']):
-        profile = profile[(profile['ELEC_dens']
-                          >= (1. / np.e) * cosmic['edmax'].iloc[i])
-                          & (profile.index >= cosmic['edmaxalt'].iloc[i])]
-        # want the first altitude where density drops below NmF2/e
-        # first, resample such that we know all altitudes in between samples
-        # are there
+        # Want the first altitude where density drops below NmF2/e.
         if len(profile) > 10:
-            i1 = profile.index[1:]
-            i2 = profile.index[0:-1]
+            i1 = profile.coords["MSL_alt"][1:]
+            i2 = profile.coords["MSL_alt"][0:-1]
             modeDiff = mode(i1.values - i2.values)[0][0]
-            profile = profile.reindex(np.arange(profile.index[0],
-                                                profile.index[-1] + modeDiff,
-                                                modeDiff))
-            # ensure there are no gaps, if so, remove all data above gap
-            idx, = np.where(profile['ELEC_dens'].isnull())
+
+            # Ensure there are no gaps, if so, remove all data above gap
+            idx, = np.where((i1.values - i2.values) > 2 * modeDiff)
+            if len(idx) > 0:
+                profile = profile[0:idx[0]]
+
+            # Ensure there are no null values in the middle of profile.
+            idx, = np.where(profile.isnull())
             if len(idx) > 0:
-                profile = profile.iloc[0:idx[0]]
+                profile = profile[0:idx[0]]
 
         if len(profile) > 10:
-            # make sure density at highest altitude is near Nm/e
-            if (profile['ELEC_dens'].iloc[-1] / profile['ELEC_dens'].iloc[0]
-                    < 0.4):
-                altDiff = profile.index.values[-1] - profile.index.values[0]
-                if altDiff >= 500:
-                    altDiff = np.nan
-                output[i] = altDiff
+            # Make sure density at highest altitude is near Nm/e
+            if profile[-1] / profile[0] < 0.4:
+                alt = profile.coords["MSL_alt"]
+                alt_diff = alt[-1] - alt["MSL_alt"][0]
+                if alt_diff >= 500:
+                    alt_diff = np.nan
+                output[i] = alt_diff
             else:
                 output[i] = np.nan
         else:
             output[i] = np.nan
 
-    return output
+    inst['thf2'] = output
+
+    return
+
 
+# Register all instruments in pysatCDAAC and pysatNASA. Only required once
+# per install.
+pysat.utils.registry.register_by_module(pysatCDAAC.instruments)
+pysat.utils.registry.register_by_module(pysatNASA.instruments)
 
-# instantiate IVM Object
-ivm = pysat.Instrument(platform='cnofs',
-                       name='ivm', tag='',
+# Dates for demo
+ssn_days = 67
+startDate = dt.datetime(2009, 12, 21) - pds.DateOffset(days=ssn_days)
+stopDate = dt.datetime(2009, 12, 21) + pds.DateOffset(days=ssn_days)
```

Comment:
Recommend changing to snake_case for all camelCase variables here.  More important here, since this is a demo that we adhere to pysat style.